src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import shutil
+import subprocess
+import requests
+import os
+import platform
+from pathlib import Path
+
+
+def aks_web_app_init():
+    filePath = binaryPreCheck()
+    if filePath:
+        isRunSuccessful = run_binary(filePath)
+        if isRunSuccessful is 0:
+            cmd_finish()
+            return
+
+    raise ValueError("Binary was NOT executed successfully")
+
+# If setup is valid this method returns the correct binary path to execute
+def binaryPreCheck() -> str:
+    print("The DraftV2 setup is in progress ...")
+
+    draftV2BinaryPath = findExistingPath(getPotentialPaths())
+    if draftV2BinaryPath:
+        return draftV2BinaryPath
+
+    print("DraftV2 binary not found")
+    return downloadBinary()
+
+
+def findExistingPath(paths) -> str:
+    print("Checking if DraftV2 binary exists ...")
+
+    operatingSystem = platform.system()
+    # Filename depends on the operating system
+    filename = "draftv2-" + operatingSystem.lower() + "-amd64"
+
+    if not paths:
+        print("List of given DraftV2 paths is empty")
+        return None
+
+    for path in paths:
+        binaryFilePath = path + "/" + filename
+        if os.path.exists(binaryFilePath):
+            print("Existing binary found at: " + binaryFilePath)
+            return binaryFilePath
+    return ""
+
+# Returns a list of potential draftV2 binary paths
+def getPotentialPaths():
+    result = [str(Path.home()) + "/" +".aksapp"]
+    paths = os.environ['PATH'].split(':')
+
+    for path in paths:
+        if "draftv2" in path:
+            result.append(path)
+    return result
+
+
+def run_binary(path):
+    if path is None:
+        raise ValueError("The given Binary path was null or empty")
+
+    print("Running DraftV2 Binary ...")
+    process = subprocess.Popen([path], stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    exit_code = process.wait()
+    print(stdout, stderr, exit_code)
+    return exit_code
+
+
+def cmd_finish():
+    # Clean up logic can go here if needed
+    print("We are done Stop.")
+
+
+def downloadBinary() -> str:
+    # prompt user to download binary. If users says no, we error out and tell them that this requires the binary
+    permissionToDownload = input("The required binary was not found. Would you like us to download the required binary for you? Y/n: ")
```

Comment:
💡 Nit, I am sure PM can really help for messaging this, also `Y/n` or `Y/N`  might be worth to check the also I noticed this, so someone can comment if `from knack.prompting import prompt_y_n` is the preferred way to use.

* https://github.com/microsoft/knack/blob/dev/docs/prompting.md 

https://github.com/Azure/azure-cli-extensions/blob/f52a3e3c361d02887ff2897b24aa6fa064a86b1a/src/aks-preview/azext_aks_preview/custom.py#L399